### PR TITLE
Fix critical issues detected with SonarCloud

### PR DESCRIPTION
**Pull request by gwenneg**
Review this PR with the `Hide whitespace changes` option enabled, it will be much shorter.

### CODE REVIEW

**Comment by gwenneg** [comment]
If you have any doubt about a specific change, I can provide a link to the SonarCloud issue.

File: extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/UndertowDeploymentRecorder.java [hunk 1]
```diff
@@ -658,9 +658,9 @@ public ByteBuf allocateBuffer() {
         @Override
         public ByteBuf allocateBuffer(boolean direct) {
             if (direct) {
-                return PartialPooledByteBufAllocator.DEFAULT.directBuffer(defaultBufferSize);
+                return ByteBufAllocator.DEFAULT.directBuffer(defaultBufferSize);
             } else {
-                return PartialPooledByteBufAllocator.DEFAULT.heapBuffer(defaultBufferSize);
+                return ByteBufAllocator.DEFAULT.heapBuffer(defaultBufferSize);
```
Comment (gsmet) [inline extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/UndertowDeploymentRecorder.java]: What's the reason of this change?

File: independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/BootstrapAppModelFactory.java [hunk 2]
```diff
@@ -445,7 +446,7 @@ private CurationResult createAppModelForJar(Path appArtifactPath) {
         final Iterator<AppDependency> depsI = appDeps.iterator();
         while (depsI.hasNext()) {
             final AppArtifact appDep = depsI.next().getArtifact();
-            if (!appDep.getType().equals(AppArtifact.TYPE_JAR)) {
+            if (!appDep.getType().equals(AppArtifactCoords.TYPE_JAR)) {
```
Comment (gsmet) [inline independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/BootstrapAppModelFactory.java]: What's the reason of this change?

**Review by gsmet — CHANGES_REQUESTED** [review]
I added a few questions. Can you have a look? Thanks!

File: independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/BootstrapAppModelFactory.java (hunk 2 above)
Comment (gwenneg) [reply independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/BootstrapAppModelFactory.java]: https://sonarcloud.io/project/issues?id=gwenneg_quarkus&open=AXAxfQvxgbHBif7sEC6s&resolutions=FIXED&severities=CRITICAL

The `TYPE_JAR` constant is actually defined in `AppArtifactCoords` which is extended by `AppArtifact`.

File: extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/UndertowDeploymentRecorder.java (hunk 1 above)
Comment (gwenneg) [reply extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/UndertowDeploymentRecorder.java]: https://sonarcloud.io/project/issues?id=gwenneg_quarkus&open=AXAxfMfPgbHBif7sECGm&resolutions=FIXED&severities=CRITICAL

The `DEFAULT` constant is actually defined in `ByteBufAllocator` which is implemented by `PartialPooledByteBufAllocator`.

**Comment by gwenneg** [comment]
I force-pushed additional changes to fix the build, sorry about that :)

**Review by gastaldi — APPROVED** [review]
LGTM

**Review by gsmet — APPROVED** [review]
Fine by me but CI is unhappy :).

**Comment by gastaldi** [comment]
Thank you!

**Comment by gwenneg** [comment]
Thanks for reviewing this!
